### PR TITLE
Installer advanced retries: docu

### PR DESCRIPTION
**Pull request by jakkab**
**Description**

Changes proposed in this pull request:

- document changes implemented in #8293 and #8407

### CODE REVIEW

File: docs/kyma/04-12-details.md [hunk 1]
```diff
@@ -5,12 +5,26 @@ type: Installation
 
 This section describes in details some of the installation mechanisms.
```
Comment (bszwarc) [inline docs/kyma/04-12-details.md:7]: I would rename the doc to `Error handling` If we have more items to add to this one we can then think of changing the name. Error handling says more than `details`.
Rename the title of the doc in the metadata and the name of the file.

File: docs/kyma/04-12-details.md [hunk 2]
```diff
@@ -5,12 +5,26 @@ type: Installation
...
 
-### Retry policy
+## Error handling
```
Comment (bszwarc) [inline docs/kyma/04-12-details.md:8]:
```suggestion

```
This one can be removed, as we have the name in the title.

File: docs/kyma/04-12-details.md [hunk 3]
```diff
@@ -5,12 +5,26 @@ type: Installation
...
+
+Kyma Operator features a retry mechanism to handle temporary issues such as Tiller unavailability or a prolonged process of creating a resource path for custom resources in the Kubernetes API Server. 
+If an error occurs while processing a component, Kyma Operator restores the initial state of that component and retries the step. Specific behavior of the controller depends on the nature of the operation that was interrupted by an error.
+
+### Installation error
```
Comment (bszwarc) [inline docs/kyma/04-12-details.md:13]:
```suggestion
## Installation error
```

File: docs/kyma/04-12-details.md [hunk 4]
```diff
@@ -5,12 +5,26 @@ type: Installation
 
 This section describes in details some of the installation mechanisms.
 
-### Retry policy
+## Error handling
+
+Kyma Operator features a retry mechanism to handle temporary issues such as Tiller unavailability or a prolonged process of creating a resource path for custom resources in the Kubernetes API Server.
```
Comment (bszwarc) [inline docs/kyma/04-12-details.md:10]:
```suggestion
Kyma Operator features a retry mechanism to handle temporary issues, such as Tiller unavailability or a prolonged process of creating a resource path for custom resources in the Kubernetes API Server. 
```

File: docs/kyma/04-12-details.md [hunk 5]
```diff
@@ -5,12 +5,26 @@ type: Installation
...
 
-### Retry policy
+## Error handling
+
+Kyma Operator features a retry mechanism to handle temporary issues such as Tiller unavailability or a prolonged process of creating a resource path for custom resources in the Kubernetes API Server. 
+If an error occurs while processing a component, Kyma Operator restores the initial state of that component and retries the step. Specific behavior of the controller depends on the nature of the operation that was interrupted by an error.
+
+### Installation error
+
+If an error occurs during the installation of a component, Kyma Operator deletes the corresponding Helm release and retries the operation. If such a release does not exist, the deletion is skipped.
```
Comment (bszwarc) [inline docs/kyma/04-12-details.md:15]:
```suggestion
If an error occurs during component installation, Kyma Operator deletes the corresponding Helm release and retries the operation. If such a release does not exist, the deletion step is skipped.
```

File: docs/kyma/04-12-details.md [hunk 6]
```diff
@@ -5,12 +5,26 @@ type: Installation
...
+
 
-Kyma Operator has a retry mechanism which makes sure that the installation does not fail in the case of temporary issues, such as Tiller unavailability or a prolonged process of creating a resource path for custom resources in the Kubernetes API Server. 
+### Upgrade error
 
-If an error occurs during the installation of a component, the installation of that particular component is retried based on the provided configuration that specifies the intervals between consecutive attempts. The default configuration consists of 5 retries with the following time intervals between each attempt: 10s, 20s, 40s, 60s, 80s. After that, the installation is stopped and can be restarted manually by setting the `action: install` label on the Installation CR.
+If an error occurs during the upgrade of a component, Kyma Operator rolls back the corresponding Helm release to the last deployed revision and retries the operation. If the history of the release does not include a deployed revision, the controller returns an error and stops the process.
```
Comment (bszwarc) [inline docs/kyma/04-12-details.md:20]:
```suggestion
If an error occurs during component upgrade, Kyma Operator rolls back the corresponding Helm release to the last deployed revision and retries the operation. If the release history does not include a deployed revision, the controller returns an error and stops the process. 
```

File: docs/kyma/04-12-details.md [hunk 7]
```diff
@@ -5,12 +5,26 @@ type: Installation
...
+### Upgrade error
 
-If an error occurs during the installation of a component, the installation of that particular component is retried based on the provided configuration that specifies the intervals between consecutive attempts. The default configuration consists of 5 retries with the following time intervals between each attempt: 10s, 20s, 40s, 60s, 80s. After that, the installation is stopped and can be restarted manually by setting the `action: install` label on the Installation CR.
+If an error occurs during the upgrade of a component, Kyma Operator rolls back the corresponding Helm release to the last deployed revision and retries the operation. If the history of the release does not include a deployed revision, the controller returns an error and stops the process. 
+
+
+### Retry policy
+ 
+Retry policy is based on the provided configuration that specifies the intervals between consecutive attempts. The default configuration consists of 5 retries with the following time intervals between each attempt: 10s, 20s, 40s, 60s, 80s. After that, the installation is stopped and can be restarted manually by setting the `action: install` label on the Installation CR.
```
Comment (bszwarc) [inline docs/kyma/04-12-details.md:25]:
```suggestion
The retry policy is based on the configuration that specifies the intervals between consecutive attempts. The default configuration consists of 5 retries with the following time intervals between each attempt: 10s, 20s, 40s, 60s, 80s. After that, the installation is stopped and can be restarted manually by setting the `action: install` label on the Installation CR. 
```

File: docs/kyma/04-12-details.md [hunk 8]
```diff
@@ -5,12 +5,26 @@ type: Installation
...
+
+### Retry policy
+ 
+Retry policy is based on the provided configuration that specifies the intervals between consecutive attempts. The default configuration consists of 5 retries with the following time intervals between each attempt: 10s, 20s, 40s, 60s, 80s. After that, the installation is stopped and can be restarted manually by setting the `action: install` label on the Installation CR. 
 
 > **NOTE:** The configuration of retries can be adjusted by setting the `backoffIntervals` argument in the Installer Deployment. The value is a comma-separated list of numbers that represent the intervals between consecutive retries. It also defines the total number of retries. For example, the default one is: `10,20,40,60,80`.
 
-In the case of an error occurring before the installation of components, the installation is retried every 30 seconds until successful. 
+
+In case of an error occurring before the installation of components, the installation is retried every 30 seconds until successful.
```
Comment (bszwarc) [inline docs/kyma/04-12-details.md:30]:
```suggestion
If an error occurs before the installation of components, the installation is retried every 30 seconds until successful.
```

File: docs/kyma/04-12-details.md (hunk 1 above)
Comment (jakkab) [reply docs/kyma/04-12-details.md]: done!

**Pull request merged**